test/Microsoft.AspNetCore.Mvc.RazorPages.Test/Internal/PageDirectiveFeatureTest.cs

```diff
@@ -7,57 +7,76 @@
 using Microsoft.AspNetCore.Razor.Evolution;
 using Xunit;
 
-namespace Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure
+namespace Microsoft.AspNetCore.Mvc.RazorPages.Internal
 {
     public class PageDirectiveFeatureTest
     {
         [Fact]
+        public void TryGetPageDirective_WithPageDirectivePrefixedWithoutAtSymbol_DoesNotParse()
+        {
+            // Arrange
+            var projectItem = new TestRazorProjectItem(@"page ""Some/Path/{value}""");
+
+            // Act & Assert
+            Assert.False(PageDirectiveFeature.TryGetPageDirective(projectItem, out var directive));
+        }
+
+        [Fact]
+        public void TryGetPageDirective_WithPageDirectiveFollowedByNoSpaces_DoesNotParse()
+        {
+            // Arrange
+            var projectItem = new TestRazorProjectItem(@"@page""Some/Path/{value}""
+The rest of the thing");
+
+            // Act & Assert
+            Assert.False(PageDirectiveFeature.TryGetPageDirective(projectItem, out var directive));
+        }
+
+        [Fact]
         public void TryGetPageDirective_FindsTemplate()
         {
             // Arrange
-            string template;
             var projectItem = new TestRazorProjectItem(@"@page ""Some/Path/{value}""
 The rest of the thing");
 
             // Act & Assert
-            Assert.True(PageDirectiveFeature.TryGetPageDirective(projectItem, out template));
-            Assert.Equal("Some/Path/{value}", template);
+            Assert.True(PageDirectiveFeature.TryGetPageDirective(projectItem, out var directive));
+            Assert.Equal("Some/Path/{value}", directive.RouteTemplate);
+            Assert.Null(directive.Name);
         }
 
         [Fact]
         public void TryGetPageDirective_NoNewLine()
         {
             // Arrange
-            string template;
             var projectItem = new TestRazorProjectItem(@"@page ""Some/Path/{value}""");
 
             // Act & Assert
-            Assert.True(PageDirectiveFeature.TryGetPageDirective(projectItem, out template));
-            Assert.Equal("Some/Path/{value}", template);
+            Assert.True(PageDirectiveFeature.TryGetPageDirective(projectItem, out var directive));
+            Assert.Equal("Some/Path/{value}", directive.RouteTemplate);
+            Assert.Null(directive.Name);
         }
 
         [Fact]
         public void TryGetPageDirective_JunkBeforeDirective()
         {
             // Arrange
-            string template;
             var projectItem = new TestRazorProjectItem(@"Not a directive @page ""Some/Path/{value}""");
 
             // Act & Assert
-            Assert.False(PageDirectiveFeature.TryGetPageDirective(projectItem, out template));
-            Assert.Null(template);
+            Assert.False(PageDirectiveFeature.TryGetPageDirective(projectItem, out var directive));
```

Comment:
Don't we want to assert something about directive here? I assume it should be null since there isn't one?